ironfish-cli/src/trusted-setup/client.ts

```diff
@@ -86,8 +86,26 @@ export class CeremonyClient {
     this.logger.error(`Server error ${ErrorUtils.renderError(error)}`)
   }
 
+  private dataBuffer = ''
+
   private onData(data: Buffer): void {
     const message = data.toString('utf-8')
+    this.dataBuffer += message
+
+    const messages = this.dataBuffer.split('\n')
+
+    if (messages.length <= 1) {
+      return
+    }
+
+    for (const message of messages.slice(0, -1)) {
+      this.onMessage(message)
+    }
+
+    this.dataBuffer = messages[message.length - 1] || ''
+  }
+
```

Comment:
You can use hugh's patented MessageBuffer here

Reply:
Nice, added